routers/api/v1/repo/commits.go

```diff
@@ -48,13 +48,61 @@ func GetSingleCommit(ctx *context.APIContext) {
 	//   "404":
 	//     "$ref": "#/responses/notFound"
 
+	sha := ctx.Params(":sha")
+	if len(sha) == 0 {
```

Comment:
I'm sure there's a function that checks if a string is a valid commit sha (i.e. accept full and shortened ones). We should perform that validation here.

Reply:
but then it would be a breaking change!